### PR TITLE
Expose local_temperature_calibration in HA as numeric

### LINKED ISSUE #9543: Expose local_temperature_calibration to Homeassistant from TS0601_thermostat
Hey,

i would like to use a script to sync the local_temperature_calibration to an external temperature sensor in Home assistant.

local_temperature_calibration is avaible via MQTT but it's not exposend as value in Homeassistant.

I would like to do this myself... but i have no idea in which file i need to start...

So mybe somebody can give me some advice with this or just add this entity

THX
This project is awesome!

### PR DESCRIPTION
**Pull request by Drafteed**
Hello! 

This PR will enable discovery a climate `local_temperature_calibration` in Home Assistant as `Number` entity.

This will close: #9543 and #9269

### CODE REVIEW

File: lib/extension/homeassistant.ts [hunk 1]
```diff
@@ -266,6 +266,24 @@ export default class HomeAssistant extends Extension {
...
+                const discoveryEntry = {
+                    type: 'number',
+                    object_id: endpoint ? `${tempCalibration.name}_${endpoint}` : `${tempCalibration.name}`,
+                    mockProperties: [tempCalibration.property],
+                    discovery_payload: {
+                        value_template: `{{ value_json.${tempCalibration.property} }}`,
+                        command_topic: true,
+                        command_topic_prefix: endpoint,
+                        command_topic_postfix: tempCalibration.property,
+                        min: -65535,
```
Comment (Koenkk) [inline lib/extension/homeassistant.ts:280]: Please use the same min/max logic as https://github.com/Koenkk/zigbee2mqtt/blob/8ef82fe23f8515a396f0178fcf6097dc137ff627/lib/extension/homeassistant.ts#L623

Comment (Drafteed) [reply lib/extension/homeassistant.ts:280]: I think done, but local temp calibration has no min/max value at all in converters. I could be wrong.

Comment (Koenkk) [reply lib/extension/homeassistant.ts:280]: That doesn't matter, we shouldn't provide these magic values, please also remove the `min` and `max` from here.

Comment (Drafteed) [reply lib/extension/homeassistant.ts]: Done :)

Comment (Drafteed) [reply lib/extension/homeassistant.ts]: P.S. I followed this values from these section:
https://github.com/Koenkk/zigbee2mqtt/blob/13986be851c5a3cabbd258117ea92d02ad80823c/lib/extension/homeassistant.ts#L565-L571

**Comment by Drafteed** [comment]
@Koenkk thanks!

**Comment by Drafteed** [comment]
@Koenkk after removing min/max magic properties HA displays it as a `range slider`, without the ability to set negative values.



calling via service with negative value throws an error:

```
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 371, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 571, in _async_call_service_step
    await service_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1495, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1530, in _execute_service
    await handler.job.target(service_call)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 213, in handle_service
    await self.hass.helpers.service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 667, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 863, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 704, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/number/__init__.py", line 64, in async_set_value
    raise ValueError(
ValueError: Value -2.0 for saswell_trv2_local_temperature_calibration is outside valid range 0.0 - 100.0
```

**Pull request merged**